autoload/test/java/maventest.vim

```diff
@@ -24,8 +24,20 @@ function! test#java#maventest#build_position(type, position) abort
 endfunction
 
 function! test#java#maventest#build_args(args) abort
-  let args = ['-Dtest='] + a:args
-  return [join(args, "")]
+  " In order to support custom options passed to mvn we need split
+  " the options from the test name (if one is passed).
+  let test_name = ''
+  let opts = []
+  let user_options = get(g:, 'test#java#maventest#options', '')
```

Comment:
Yes and that's inherently _the issue_ here.

For maven we need to insert a cli parameter `-DTest=` before the test name but after any custom options the user gives, so I'm comparing which items in `a:args` are also passed as options, and which (if any) is the test name. 

I initially had something like this (pardon my poor vimscript knowledge):
```
let args = a:args[:-2] + ['-Dtest=' . a:args[-1]]
```
The above takes all items in `a:args` until the last one (which I assume is the test file) and appends a concatenation of `-Dtest=TestName`. If we use the above and want to pass `-f /path/to/pom.xml` for mvn it will work with `:TestFile` and  `:TestNearest`. It would break `:TestSuite` though as it would eventually result in
```
$ mvn -f /path/to/pom.xml -Dtest=/path/to/pom.xml
```

I haven't really worked with vimscript before so there might be a better way to achieve this. Suggestions are welcome.